### PR TITLE
Use helm repo for integration tests instead of tarballs

### PR DESCRIPTION
**Pull request by kfaseela**
Signed-off-by: Faseela K <faseela.k@est.tech>

Make the helm integration tests use istio helm repo.
This is needed as part of helm graduation from Alpha to Beta (https://github.com/istio/istio/issues/42565)

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
I like the simplicity but am a bit concerned it goes the opposite direction of https://github.com/istio/istio/issues/35835. WDYT?

Since its GCS it may not be so bad since we already heavily depend on it

**Comment by kfaseela** [comment]
> I like the simplicity but am a bit concerned it goes the opposite direction of #35835. WDYT?
> 
> Since its GCS it may not be so bad since we already heavily depend on it

Ohh!! I missed this discussion. I thought we wanted to get rid of tarballs based install everywhere as we discussed in the helm promotion task, so thought this was also needed!

**Comment by jacob-delgado** [comment]
If we don't test the helm repo then we need to figure out a way to make updating the tarballs and helm integration test. Right now it's a bit difficult to update them. It'd be nice if it was part of the release process, but it's too cumbersome to do for a release manager without understanding how they are packaged and tested.

File: pkg/test/helm/helm.go [hunk 1]
```diff
@@ -35,6 +35,12 @@ func New(kubeConfig string) *Helm {
 	}
 }
 
+// InstallChartWithValues installs the specified chart with its given name to the given namespace
+func (h *Helm) AddRepo() error {
+	_, err := execCommand("helm repo add istio https://istio-release.storage.googleapis.com/charts")
```
Comment (keithmattix) [inline pkg/test/helm/helm.go:40]: Could we make this configurable so folks could use this test suite with their own chart repo?

Comment (kfaseela) [reply pkg/test/helm/helm.go]: Done

**Comment by kfaseela** [comment]
> If we don't test the helm repo then we need to figure out a way to make updating the tarballs and helm integration test. Right now it's a bit difficult to update them. It'd be nice if it was part of the release process, but it's too cumbersome to do for a release manager without understanding how they are packaged and tested.

@howardjohn what do you think?

File: pkg/test/helm/helm.go (hunk 1 above)
Comment (keithmattix) [reply pkg/test/helm/helm.go]: Wonderful; thank you!

**Comment by keithmattix** [comment]
> I like the simplicity but am a bit concerned it goes the opposite direction of https://github.com/istio/istio/issues/35835. WDYT?
> 
> 
> 
> Since its GCS it may not be so bad since we already heavily depend on it

With the configurable chart repo, a local web server that hosts the charts can likely be added later right?

**Comment by howardjohn** [comment]
yep lgtm (at ahigh level, will review the code in a bit)

File: pkg/test/helm/helm.go [hunk 2]
```diff
@@ -35,6 +36,21 @@ func New(kubeConfig string) *Helm {
 	}
 }
 
+// Add Repo adds the specified helm repo to use for installing the charts
+func (h *Helm) AddRepo(t framework.TestContext) {
+	command := fmt.Sprintf("helm repo add istio %s", t.Settings().HelmRepo)
```
Comment (howardjohn) [inline pkg/test/helm/helm.go:41]: can we/should we just add `--repo HelmRepo` to each command? The current command will modify users global helm settings (and may conflict if they have `istio` defined. Especially DeleteRepo

Comment (kfaseela) [reply pkg/test/helm/helm.go:41]: Makes sense. Will fix

**Comment by nshankar13** [comment]
@kfaseela as of now, what is the current way of running the helm e2e tests in `/tests/integration/helm` against helm chart tarballs that are built and packaged locally? 

For instance, if we had the tarballs for `base-1.16.1.tgz  cni-1.16.1.tgz  gateway-1.16.1.tgz  istiod-1.16.1.tgz` living in some `/dir`, how do we point the e2es to install / test against these charts?

File: pkg/test/helm/helm.go (hunk 2 above)
Comment (kfaseela) [reply pkg/test/helm/helm.go]: @howardjohn done

**Comment by kfaseela** [comment]
> @kfaseela as of now, what is the current way of running the helm e2e tests in `/tests/integration/helm` against helm chart tarballs that are built and packaged locally?
> 
> For instance, if we had the tarballs for `base-1.16.1.tgz cni-1.16.1.tgz gateway-1.16.1.tgz istiod-1.16.1.tgz` living in some `/dir`, how do we point the e2es to install / test against these charts?

currently the tarballs are hardcoded as far as i can understand, and there is no way to override. If you replace the tarballs with your own, under the testdata folder, and run the tests it will be picked up

**Comment by nshankar13** [comment]
@kfaseela thank you!!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**